### PR TITLE
[vim] Don't throw errors to signal missing options

### PR DESCRIPTION
**Pull request by vincentwoo**
Addresses #4771

### CODE REVIEW

File: keymap/vim.js [hunk 1]
```diff
@@ -411,11 +411,11 @@
       cfg = cfg || {};
       var scope = cfg.scope;
       if (!option) {
-        throw Error('Unknown option: ' + name);
+        return new Error('Unknown option: ' + name);
```
Comment (adrianheine) [inline keymap/vim.js]: Why do you not use `throw` and `catch` for this?

Comment (vincentwoo) [inline keymap/vim.js]: Doesn't really seem necessary. We're just trying to signal that an option was missing. We could return a hash signalling error, but this changes the least of the original code without requiring try/catch blocks.

Comment (marijnh) [inline keymap/vim.js]: I agree -- exception control flow, especially in JavaScript, is quite a lot harder to deal with and easier to mess up, so a regular return seems sensible.

**Comment by marijnh** [comment]
Merged as 27de924e

**Pull request closed without merging**